### ISSUE TITLE
how to get output "moved -> moving"

### ISSUE THREAD

**Issue by eremo2002** [voxelmorph/voxelmorph#445]
Hi, I trained VoxelMorph for deformable registration.
I got "moved data" using input (moving, fixed).
But, I want result "moved -> moving". Is it possible using voxelmorph?
How to get inverse result? (moved -> moving).

Thank you.

**Comment by jiwoosong** [comment]
I just found a way to solve this problem. Well, first of all, in my case, it may be slightly different from your problem because I am using a combination of DiffeoMorphic and Cycle Morph. Therefore, I will show you a simple way to reverse the `torch.nn.functional.grid_sample `function. For convenience, it is abbreviated as follows.

`moving_img `: original voxel (X, Y, Z)
`norm_coord`: Normalized coordinates (3, X, Y, Z)
`disp_field`: displacement vector to be shifted from normalized coordinates (3, X, Y, Z)
`deform_field`: Coordinates on the `norm_coord` to be moved at each position (3, X, Y, Z)

If you use torch.nn.functional.grid_sample as is, the input to the function will be `m_img` 5D (B, 1, X, Y, Z), `deform_field` (B, 3, X, Y, Z). `deform_field` is the output of VoxelMorph and is the normalized coordinates of the target to be warped backward. It has a size between [-1, 1]. `moving_img` can be converted to `moved_img` as shown below.

``` python
moved_img = grid_sample(moving_img, deform_field)
```

In order to perform the inverse transformation, it is necessary to track the displacement and return the vector in the opposite direction.

``` python
disp_field = deform_field - norm_coord
inv_disp_field = - grid_sample(disp_field, deform_field)
inv_deform_field = inv_disp_field + norm_coord
reversed_img = grid_sample(moved_img, inv_deform_field)
```
It's simple to use, but the inverse transform where it's clipped or there's no pixel mapping needs further improvement.

**Comment by adalca** [comment]
Thank s @jiwoosong 
@eremo2002 does this help? Are you using tf or pytorch?

**Comment by eremo2002** [comment]
Thanks a lot! It is helpful example.
I'll try it.
